app/models/concerns/audit_extensions.rb

```diff
@@ -33,9 +34,29 @@ def ensure_taxonomies_not_escalated
   private
 
   def log_audit
-    Foreman::Logging.with_fields(self.audited_changes) do
-      Foreman::Logging.logger('audit').info { "#{self.action} event for #{self.auditable_type} with id #{self.auditable_id}" }
+    telemetry_increment_counter(:audit_records_created, 1, type: self.auditable_type)
+    audit_logger = Foreman::Logging.logger('audit')
+    return unless (self.audited_changes && audit_logger.info?)
+    self.audited_changes.each_pair do |column_name, old_and_new|
+      audited_fields = {
+        audit_action: self.action,
+        audit_type: self.auditable_type,
+        audit_id: self.auditable_id,
+        audit_column: column_name
+      }
+      if old_and_new.is_a?(Array)
```

Comment:
this should check if `self.action==update`, not if the change is an array - in case the value itself is an array.